src/rdkafka_conf.c

```diff
@@ -1135,6 +1135,21 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
      "Group protocol type. NOTE: Currently, the only supported group "
      "protocol type is `consumer`.",
      .sdef = "consumer"},
+    {_RK_GLOBAL | _RK_CGRP | _RK_HIGH, "group.protocol", _RK_C_S2I,
+     _RK(group_protocol),
+     "Group protocol to use. Use `generic` for the old protocol "
```

Comment:
Apply the documentation changes here and run `make CONFIGURATION.md` to regenerate the file.